dss_datamover/logger.py

```diff
@@ -141,79 +167,83 @@ def logging(self, queue, lock, stop_logging):
-                        fh.write(str(time.ctime()) + ": " + LOGGING_LEVEL[message_level] + ": " + message_value + "\n")
-                    elif message is None:
-                        stop_flag = True
-                        break
-                    else:
-                        fh.write(str(time.ctime()) + ": " + message + "\n")
+            logger_handle_thr = logging.getLogger(app_name)
+            logger_handle_thr.setLevel(log_level)
+        except Exception as e:
+            print('Exception in creating logger handle inside the logger thread')
+            raise
 
-                if stop_logging.value:
-                    if stop_flag and queue.qsize() and stop_counter < stop_counter_threshold:
+        try:
+            rfh = self.create_rotating_file_handler(log_file, self.max_log_file_size, self.log_file_backup_count)
+            logger_handle_thr.addHandler(rfh)
+            logger_handle_thr.addHandler(StreamHandler())
+
+            while True:
+                message = queue.get()
+                if message is None:
+                    stop_flag = True
+                else:
+                    logger_handle_thr.handle(message)
+
+                if stop_flag:
+                    if queue.qsize() and stop_counter < stop_counter_threshold:
                         print("Queue is not empty, but received a stop signal ...")
                         stop_counter += 1
+                        time.sleep(1)
                     else:
                         break
-                time.sleep(1)
 
         except Exception as e:
-            print("Exception: {}".format(e))
-        finally:
-            if fh:
-                fh.close()
+            print("Exception in logger process: {}".format(e))
 
     @exception
     def info(self, message):
-        msg = (0, message)
-        self.queue.put(msg)
+        self.logger_handle.info(message)
 
     @exception
     def debug(self, message):
-        if self.logging_level < len(LOGGING_LEVEL) and LOGGING_LEVEL[self.logging_level] == "DEBUG":
-            msg = (1, message)
-            self.queue.put(msg)
+        self.logger_handle.debug(message)
 
     @exception
     def warn(self, message):
-        if self.logging_level <= 2:
-            msg = (2, message)
-            self.queue.put(msg)
+        self.logger_handle.warning(message)
 
     @exception
     def error(self, message):
-        if self.logging_level <= 3:
-            msg = (3, message)
-            self.queue.put(msg)
+        self.logger_handle.error(message)
 
     @exception
     def excep(self, message):
-        if self.logging_level <= 4:
-            msg = (4, message)
-            self.queue.put(msg)
+        self.logger_handle.exception(message)
 
     @exception
     def fatal(self, message):
-        if self.logging_level <= 5:
-            msg = (5, message)
-            self.queue.put(msg)
+        self.logger_handle.fatal(message)
+
+
+if __name__ == "__main__":
```

Comment:
Please remove this testing block

Reply:
This is not just testing block. Usually you need to have such code so that the particular file can be run without any extra validation.

Reply:
Can we move this to be a testcase within pytest framework under tests/ folder. Specifically, in the tests/test_logger.py file?